pages/folder.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {
+  Dispatch,
+  SetStateAction,
+  createContext,
+  useEffect,
+  useState,
+} from "react";
+import styled from "styled-components";
+import { getAllLinks, getFolderLinks, getFolders } from "@/src/api";
+import SearchBar from "@/src/components/common/SearchBar";
+import AddLink from "@/src/components/folder/AddLink";
+import FolderInfo from "@/src/components/folder/FolderInfo";
+import useAsync from "@/src/components/hooks/useAsync";
+import CardList from "@/src/components/shared/CardList";
+
+const Container = styled.main`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  background-color: white;
+  padding-bottom: 100px;
+`;
+
+export const FolderContext = createContext<{
+  folders: any[];
+  folderId: number;
+  setFolderId: Dispatch<SetStateAction<number>>;
+}>({ folders: [], folderId: 0, setFolderId: () => {} });
+
+function Folder() {
+  const [folderId, setFolderId] = useState<number>(0);
+  const [folders, setFolders] = useState<any[]>([1]);
+  const [links, setLinks] = useState<any[]>([]);
+  const [allLinks, setAllLinks] = useState<any[]>([]);
+  const [inputValue, setInputValue] = useState<string>("");
+
+  const [, getFoldersAsync] = useAsync(getFolders);
+  const [, getFolderLinksAsync] = useAsync(getFolderLinks);
+  const [, getAllLinksAsync] = useAsync(getAllLinks);
+
+  const fetchData = async (folderId: number) => {
+    try {
+      const promises: Promise<any>[] = [];
+
+      if (typeof getFoldersAsync === "function")
+        promises.push(getFoldersAsync());
+      if (typeof getAllLinksAsync === "function")
+        promises.push(getAllLinksAsync());
+      if (typeof getFolderLinksAsync === "function")
+        promises.push(getFolderLinksAsync(folderId));
+
+      const [foldersResponse, allLinksResponse, folderLinksResponse] =
+        await Promise.all(promises);
+
+      if (foldersResponse) setFolders(foldersResponse.data);
+      if (allLinksResponse) setAllLinks(allLinksResponse.data);
+      if (folderLinksResponse) setLinks(folderLinksResponse.data);
+    } catch (error) {
+      console.error("Error:", error);
+    }
+  };
```

Comment:
여기서 `useAsync`는 항상 [Error | null, 함수] 형태의 튜플을 반환하고 있는데요, 두 번째 요소는 항상 함수 타입이며, 이 함수는 asyncFunction의 비동기 실행을 감싸고 있습니다. 따라서 typeof getFoldersAsync === "function" 같은 체크는 대부분 필요하지 않을 것 같아요. 

<필요한 경우>
예를들어 `getFoldersAsync` 가 정의되지 않았거나 변경될 수 있는 경우인데, getFoldersAsync가 동적으로 할당되거나 다른 유형의 값으로 변경될 수 있다면, typeof getFoldersAsync === "function"  타입 체크는 유효할거에요. 그러나 일반적으로 useAsync 훅을 사용할 때는 함수가 항상 반환되므로, 이런 상황은 드뭅니다.


결과적으로 아래처럼 변경해볼 수 있을 것 같아요.

```suggestion

const fetchData = async (folderId: number) => {
  try {
    const folderPromise = getFoldersAsync();
    const linksPromise = getAllLinksAsync();
    const folderLinksPromise = getFolderLinksAsync(folderId);

    const [foldersResponse, allLinksResponse, folderLinksResponse] =
      await Promise.all([folderPromise, linksPromise, folderLinksPromise]);

    setFolders(foldersResponse?.data);
    setAllLinks(allLinksResponse?.data);
    setLinks(folderLinksResponse?.data);
  } catch (error) {
    // UI에 에러 표시 또는 적절한 에러 처리
  }
};

```